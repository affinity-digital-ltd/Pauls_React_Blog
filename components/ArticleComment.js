import { Component } from 'react'

import CommentForm from '../components/CommentForm'
import Auth from '../lib/Auth'

class ArticleComment extends Component {
  constructor (props) {
    super(props)

    this.auth = new Auth()
    this.login = this.login.bind(this)
    this.logout = this.logout.bind(this)
  }

  login () {
    this.auth.login()
  }

  userProfile () {
    return {
      name: localStorage.getItem('name')
    }
  }

  logout () {
    this.auth.logout()
  }

  render () {
    const userProfile = this.userProfile()
    if (this.auth.isAuthenticated()) {
      return <div>
        <div className='c-user-comments__info'>
          <h3 className='c-user-comments__login-message'>Welcome, {userProfile.name}</h3>
          <a className='c-user-comments__login-btn' onClick={this.logout}>Logout</a>
        </div>
        <CommentForm />
      </div>
    } else {
      return <div className='c-user-comments__info'>
        <h3 className='c-user-comments__login-message'>Please login to comment.</h3>
        <a className='c-user-comments__login-btn' onClick={this.login} >Login</a>
      </div>
    }
  }
}

export default ArticleComment
