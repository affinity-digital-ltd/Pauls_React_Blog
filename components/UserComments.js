import { Component } from 'react'

import Comment from './Comment'
import ArticleComment from './ArticleComment'

import Pluralize from 'react-pluralize'

class UserComments extends Component {
  render () {
    const { comments } = this.props
    return <div className='c-user-comments'>
      <h2 className='c-article__header'>Post a comment!</h2>
      <div className='c-header-divider' />
      <ArticleComment />
      <h4 className='c-user-comments__counter'><Pluralize singular={'comment'} plural={'comments'} count={comments.length} /></h4>
      {comments.map(comment => {
        return <Comment comment={comment} key={comment.id} />
      })}
    </div>
  }
}

export default UserComments
