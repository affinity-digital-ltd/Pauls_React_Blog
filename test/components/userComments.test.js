import UserComments from '../../components/UserComments'
import ArticleComment from '../../components/ArticleComment'

describe('<UserComments  />', () => {
  it('it renders correctly', () => {
    const comments = [{
      id: 'ID',
      slug: 'Title',
      createdAt: '2018-12-12 10:20:46 UTC',
      authorName: 'authorName',
      body: 'This is a commment!'
    }]

    const postComment = [{
      id: 'ID',
      slug: 'Title',
      createdAt: '2018-12-12 10:20:46 UTC',
      authorName: 'authorName',
      body: 'This is a commment!'
    }]

    const component = shallow(
      <div>
        <UserComments comments={comments} />
        <ArticleComment postComment={postComment} />
      </div>
    )

    expect(component).toMatchSnapshot()
  })

  it('should render correctly without a comment', () => {
    const comments = []
    const component = shallow(
      <UserComments comments={comments} />
    )

    expect(component).toMatchSnapshot()
  })
})
