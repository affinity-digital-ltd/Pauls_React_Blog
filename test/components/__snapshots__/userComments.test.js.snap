// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserComments  /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <UserComments
      comments={
        Array [
          Object {
            "authorName": "authorName",
            "body": "This is a commment!",
            "createdAt": "2018-12-12 10:20:46 UTC",
            "id": "ID",
            "slug": "Title",
          },
        ]
      }
    />
    <ArticleComment
      postComment={
        Array [
          Object {
            "authorName": "authorName",
            "body": "This is a commment!",
            "createdAt": "2018-12-12 10:20:46 UTC",
            "id": "ID",
            "slug": "Title",
          },
        ]
      }
    />
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <UserComments
          comments={
            Array [
              Object {
                "authorName": "authorName",
                "body": "This is a commment!",
                "createdAt": "2018-12-12 10:20:46 UTC",
                "id": "ID",
                "slug": "Title",
              },
            ]
          }
        />,
        <ArticleComment
          postComment={
            Array [
              Object {
                "authorName": "authorName",
                "body": "This is a commment!",
                "createdAt": "2018-12-12 10:20:46 UTC",
                "id": "ID",
                "slug": "Title",
              },
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "comments": Array [
            Object {
              "authorName": "authorName",
              "body": "This is a commment!",
              "createdAt": "2018-12-12 10:20:46 UTC",
              "id": "ID",
              "slug": "Title",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "postComment": Array [
            Object {
              "authorName": "authorName",
              "body": "This is a commment!",
              "createdAt": "2018-12-12 10:20:46 UTC",
              "id": "ID",
              "slug": "Title",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <UserComments
            comments={
              Array [
                Object {
                  "authorName": "authorName",
                  "body": "This is a commment!",
                  "createdAt": "2018-12-12 10:20:46 UTC",
                  "id": "ID",
                  "slug": "Title",
                },
              ]
            }
          />,
          <ArticleComment
            postComment={
              Array [
                Object {
                  "authorName": "authorName",
                  "body": "This is a commment!",
                  "createdAt": "2018-12-12 10:20:46 UTC",
                  "id": "ID",
                  "slug": "Title",
                },
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "comments": Array [
              Object {
                "authorName": "authorName",
                "body": "This is a commment!",
                "createdAt": "2018-12-12 10:20:46 UTC",
                "id": "ID",
                "slug": "Title",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "postComment": Array [
              Object {
                "authorName": "authorName",
                "body": "This is a commment!",
                "createdAt": "2018-12-12 10:20:46 UTC",
                "id": "ID",
                "slug": "Title",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<UserComments  /> should render correctly without a comment 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserComments
    comments={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="c-article__header"
        >
          Post a comment!
        </h2>,
        <div
          className="c-header-divider"
        />,
        <ArticleComment
          postComment={undefined}
        />,
        <h4
          className="c-user-comments__counter"
        >
          <Plural
            className={null}
            count={0}
            plural="comments"
            showCount={true}
            singular="comment"
            style={Object {}}
            zero={null}
          />
        </h4>,
      ],
      "className": "c-user-comments",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Post a comment!",
          "className": "c-article__header",
        },
        "ref": null,
        "rendered": "Post a comment!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "c-header-divider",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "postComment": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Plural
            className={null}
            count={0}
            plural="comments"
            showCount={true}
            singular="comment"
            style={Object {}}
            zero={null}
          />,
          "className": "c-user-comments__counter",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": null,
            "count": 0,
            "plural": "comments",
            "showCount": true,
            "singular": "comment",
            "style": Object {},
            "zero": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="c-article__header"
          >
            Post a comment!
          </h2>,
          <div
            className="c-header-divider"
          />,
          <ArticleComment
            postComment={undefined}
          />,
          <h4
            className="c-user-comments__counter"
          >
            <Plural
              className={null}
              count={0}
              plural="comments"
              showCount={true}
              singular="comment"
              style={Object {}}
              zero={null}
            />
          </h4>,
        ],
        "className": "c-user-comments",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Post a comment!",
            "className": "c-article__header",
          },
          "ref": null,
          "rendered": "Post a comment!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-header-divider",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "postComment": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Plural
              className={null}
              count={0}
              plural="comments"
              showCount={true}
              singular="comment"
              style={Object {}}
              zero={null}
            />,
            "className": "c-user-comments__counter",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": null,
              "count": 0,
              "plural": "comments",
              "showCount": true,
              "singular": "comment",
              "style": Object {},
              "zero": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
